### PR TITLE
undo delete

### PR DESCRIPTION
**Pull request by acrosca**
- [x] I carefully read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md) and agree to them.

UX for #1430

### CODE REVIEW

**Comment by theScrabi** [comment]
> UX for #1430

Muchas gracias :)

Although be carefull UX stands for "User Experience", you might want to say UI.

File: app/src/main/java/org/schabi/newpipe/download/DeleteManager.java [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+import io.reactivex.Completable;
+import io.reactivex.Observable;
+import io.reactivex.android.schedulers.AndroidSchedulers;
+import io.reactivex.disposables.Disposable;
+import io.reactivex.schedulers.Schedulers;
+import io.reactivex.subjects.PublishSubject;
+import us.shandian.giga.get.DownloadManager;
+import us.shandian.giga.get.DownloadMission;
+
+public class DeleteManager {
```
Comment (theScrabi) [inline app/src/main/java/org/schabi/newpipe/download/DeleteManager.java]: Maybe it was better to call this DeleteDownloadManager

File: app/src/main/java/org/schabi/newpipe/download/DeleteManager.java [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+        if (mPendingMap.size() == 1) {
+            showUndoDeleteSnackbar(mission);
+        }
+    }
+
+    public void setDownloadManager(@NonNull DownloadManager downloadManager) {
+        mDownloadManager = downloadManager;
+
+        if (mPendingMap.size() < 1) {
+            //nothing to do
```
Comment (theScrabi) [inline app/src/main/java/org/schabi/newpipe/download/DeleteManager.java]: I think it's clear from the code that you want to abord the call here, so no need to commend it.

You could shorten it like this: `if(mPendingMap.size() < 1) return;`.

File: app/src/main/java/org/schabi/newpipe/download/DeleteManager.java [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+                showUndoDeleteSnackbar();
+            }
+        });
+
+        snackbar.show();
+    }
+
+    public void deletePending() {
+        if (mPendingMap.size() < 1) {
+            // nothing to do
```
Comment (theScrabi) [inline app/src/main/java/org/schabi/newpipe/download/DeleteManager.java]: I think it's clear from the code that you want to abord the call here, so no need to commend it.

You could shorten it like this: `if(mPendingMap.size() < 1) return;`.

File: app/src/main/java/org/schabi/newpipe/download/DeleteManager.java [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+            //nothing to do
+            return;
+        }
+
+        showUndoDeleteSnackbar();
+    }
+
+    public void restoreState(@Nullable Bundle savedInstanceState) {
+        if (savedInstanceState == null) {
+            // nothing to do
```
Comment (theScrabi) [inline app/src/main/java/org/schabi/newpipe/download/DeleteManager.java]: Same here

File: app/src/main/java/org/schabi/newpipe/download/DeleteManager.java [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+
+        List<String> list = savedInstanceState.getStringArrayList(KEY_STATE);
+        if (list != null) {
+            mPendingMap.addAll(list);
+        }
+    }
+
+    public void saveState(@Nullable Bundle outState) {
+        if (outState == null) {
+            // nothing to do
```
Comment (theScrabi) [inline app/src/main/java/org/schabi/newpipe/download/DeleteManager.java]: Same here

File: app/src/main/java/org/schabi/newpipe/download/DeleteManager.java [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+        for (Disposable disposable : mDisposableList) {
+            disposable.dispose();
+        }
+
+        outState.putStringArrayList(KEY_STATE, new ArrayList<>(mPendingMap));
+    }
+
+    private void showUndoDeleteSnackbar() {
+        if (mPendingMap.size() < 1) {
+            // nothing to do
```
Comment (theScrabi) [inline app/src/main/java/org/schabi/newpipe/download/DeleteManager.java]: Same here

File: app/src/main/java/org/schabi/newpipe/download/DeleteManager.java (hunk 3 above)
Comment (theScrabi) [inline app/src/main/java/org/schabi/newpipe/download/DeleteManager.java]: Same here

File: app/src/main/java/org/schabi/newpipe/download/DeleteManager.java [hunk 7]
```diff
@@ -0,0 +1,178 @@
...
+        for (Integer id : idSet) {
+            mDownloadManager.deleteMission(id);
+        }
+
+        mPendingMap.clear();
+    }
+
+    private void deletePending(@NonNull DownloadMission mission) {
+        if (!contains(mission)) {
+            // nothing to do
```
Comment (theScrabi) [inline app/src/main/java/org/schabi/newpipe/download/DeleteManager.java]: Same here

**Comment by acrosca** [comment]
fixed and updated

File: app/src/main/java/org/schabi/newpipe/download/DeleteDownloadManager.java [hunk 8]
```diff
@@ -0,0 +1,158 @@
...
+import io.reactivex.Completable;
+import io.reactivex.Observable;
+import io.reactivex.android.schedulers.AndroidSchedulers;
+import io.reactivex.disposables.Disposable;
+import io.reactivex.schedulers.Schedulers;
+import io.reactivex.subjects.PublishSubject;
+import us.shandian.giga.get.DownloadManager;
+import us.shandian.giga.get.DownloadMission;
+
+public class DeleteDownloadManager {
```
Comment (theScrabi) [inline app/src/main/java/org/schabi/newpipe/download/DeleteDownloadManager.java]: Maybe this should be called DeleteDownloadsManager. In order to reduce confusion.

**Comment by justanidea** [comment]
Hey
when i turn my phone to landscape, the video i just deleted reappears xD
And if i delete it and then close the app, the video is still in my files, not deleted at all...something is wrong here, sorry to bring bad news...

**Comment by justanidea** [comment]
* whereas if i go out of the download section with the arrow, everything is deleted as expected

**Comment by acrosca** [comment]
It woks as designed. In this case, I had to decide to delete or not delete when user could not undo the action. I chose not to delete.

I am not surprised that if you exit the app the file is not deleted. 

Screen rotation on the other hand should reshow the undo option and delete the file.

**Pull request merged**